doc/examples/reconst_ivim.py

```diff
@@ -0,0 +1,211 @@
+"""
+============================================================
+Intravoxel incoherent motion
+============================================================
+The intravoxel incoherent motion (IVIM) model describes diffusion
+and perfusion in the signal acquired with diffusion. The IVIM model
+can be understood as an adaptation of the work of Stejskals and Tanners
+[Stejskal65]_ in biologic tissue, and was proposed by Le Bihan [LeBihan84]_
+The model posits that two compartments exist: a slow moving compartment,
+where particles diffuse in a Brownian fashion as a consequence of thermal
+energy, and a fast moving compartment (the vascular compartment), where
+blood moves as a consequence of pressure gradient. In this second
+compartment, a pseudo diffusion term (D* ) is introduced that describes
+the displacement of the blood elements in an assumed randomly laid out
+vascular network, at the macroscopic level. For the perfusion to have a
+physiological meaning, one expects that D* is greater than D.
+
+The IVIM model expresses the diffusion signal as follows:
+
+ .. math: :
+
+    S(b) = S_{0}[fe ^ {-b*D_star} + (1 - f)e^{-b*D}]
+
+where $\mathbf{b}$ is the gradient value (which is dependent on
+the measurement parameters), $S_0$ is the signal in the absence
+of diffusion gradient sensitization, $\mathbf{f}$ is the perfusion
+fraction, $\mathbf{D}$ is the diffusion coefficient,
+ $\mathbf{D * }$ is the pseudo-diffusion diffusion constant,
+ due to vascular contributions
+
+In the following example we show how to fit the IVIM model
+on a diffusion weighted dataset and visualize the perfusion
+and diffusion. First, we import all relevant modules:
+"""
+
+import matplotlib.pyplot as plt
+
+from dipy.reconst.ivim import IvimModel, ivim_function
+from dipy.data.fetcher import read_ivim
+
+"""
+We get an IVIM dataset using Dipy's data fetcher `read_ivim`.
+This dataset was acquired with 21 b-values in 3 different directions.
+Volumes corresponding to different directions were registered to each
+other, and averaged across directions. Thus, this dataset has 4 dimensions,
+with the length of the last dimension corresponding to the number
+of b-values.
+"""
+
+img, gtab = read_ivim()
+
+"""
+img contains a nibabel Nifti1Image object(with the data)
+and gtab contains a GradientTable object(information about
+the gradients e.g. b-values and b-vectors). We get the
+data from img using `read_data`.
+"""
+
+data = img.get_data()
+print('data.shape (%d, %d, %d, %d)' % data.shape)
+
+"""
+The data has 54 slices, with 256-by-256 voxels in each slice.
+The fourth dimension corresponds to the b-values in the gtab.
+Let us visualize the data by taking a slice midway(z=27) at bvalue = 0.
```

Comment:
It says "27" here, but the code refers to `z=33`. Please correct this in the text!